pkg/questionModel/questionModel.go

```diff
@@ -0,0 +1,208 @@
+	fmt.Println()
+	model.InitializeModel(questionData)
+	p := tea.NewProgram(model)
+	if err := p.Start(); err != nil {
+		fmt.Printf("Alas, there's been an error: %v", err)
+		os.Exit(1)
+	}
+	return model.getError()
+}
+
+func createItems(data *BubbleTeaData) (header string, itemList []list.Item, itemMap map[item]string) {
+	tableString := createQuestionTable(data.OptionData, data.HeaderStrings)
+	optionStrings := strings.Split(strings.TrimSuffix(tableString, "\n"), "\n")
+
+	// Remove Empty Lines
+	for index := 0; index < len(optionStrings); index++ {
+		optionString := optionStrings[index]
+		if strings.TrimSpace(optionString) == "" {
+			optionStrings = append(optionStrings[0:index], optionStrings[index+1:]...)
+			index--
+		}
+	}
+
+	header = ""
+	if len(data.HeaderStrings) != 0 {
+		header = strings.Join(optionStrings[0:2], "\n")
+		optionStrings = optionStrings[2:]
+	}
+
+	itemList = []list.Item{}
+	itemMap = make(map[item]string, len(data.OptionData))
+	for index, itemString := range optionStrings {
+		if len(data.IndexedOptions) == len(data.OptionData) {
+			itemMap[item(itemString)] = data.IndexedOptions[index]
+		}
+		if strings.TrimSpace(itemString) != "" {
+			itemList = append(itemList, item(itemString))
+		}
+
+	}
+
+	return header, itemList, itemMap
+}
+
+func createModelList(items []list.Item, itemDelegate itemDelegate, defaultOptionIndex int) list.Model {
+	modelList := list.New(items, itemDelegate, defaultWidth, len(items)+1)
+	modelList.SetShowStatusBar(false)
+	modelList.SetFilteringEnabled(false)
+	modelList.SetShowTitle(false)
+	modelList.Styles.HelpStyle = helpStyle
+	modelList.SetShowPagination(false)
+	modelList.Select(defaultOptionIndex)
+	modelList.DisableQuitKeybindings()
+	modelList.SetShowHelp(false)
+	return modelList
+}
+
+func createQuestionTable(tableData [][]string, headers []string) string {
+	if len(headers) != 0 {
+		headerSeperators := []string{}
+		for index := 0; index < len(headers); index++ {
+			headerSeperators = append(headerSeperators, "---")
+		}
+		tableData = append([][]string{headerSeperators}, tableData...)
+	}
+
+	tableBuilder := &strings.Builder{}
+	table := tablewriter.NewWriter(tableBuilder)
+	table.SetHeader(headers)
+	table.SetAutoWrapText(false)
+	table.SetAutoFormatHeaders(true)
+	table.SetHeaderAlignment(tablewriter.ALIGN_LEFT)
+	table.SetAlignment(tablewriter.ALIGN_LEFT)
+	table.SetCenterSeparator("")
+	table.SetColumnSeparator("|")
+	table.SetRowSeparator("")
+	table.SetHeaderLine(false)
+	table.SetBorder(false)
+	table.SetTablePadding("")   // pad with tabs
+	table.AppendBulk(tableData) // Add Bulk Data
+	table.Render()
+
+	tableString := tableBuilder.String()
+
+	if len(headers) != 0 {
+		renderer, err := glamour.NewTermRenderer(glamour.WithStylePath("notty"))
+		if err == nil {
+			tableString, err = renderer.Render(tableString)
+		}
+	}
+
+	return tableString
+}
+
+func focusTableItem(tableItem string) string {
+	splitString := strings.Split(tableItem, "│")
```

Comment:
It might be a good idea to create a constant for the column separator, since it's used in several places.